process/interface.go

```diff
@@ -41,6 +41,30 @@ type HeaderValidator interface {
 	IsHeaderValidForProcessing(headerHandler data.HeaderHandler) bool
 }
 
+// InterceptedDataFactory can create new instances of InterceptedData
+type InterceptedDataFactory interface {
+	Create(buff []byte) (InterceptedData, error)
+}
+
+// InterceptedData represents the interceptor's view of the received data
+type InterceptedData interface {
+	CheckValid() error
+	IsAddressedToOtherShards() bool
+}
+
+// InterceptorProcessor processes received data
+type InterceptorProcessor interface {
+	CheckValidForProcessing(data InterceptedData) error
+	ProcessInteceptedData(data InterceptedData) error
```

Comment:
Process is used to actually modify the state trie. Interceptor pre-validates and saves the data into the datapools. SaveData() ?

Reply:
How about this:
```GO
// InterceptorWorker processes received data (usually from an interceptor)
type InterceptorWorker interface {
	Validate(data InterceptedData) error
	Save(data InterceptedData) error
}
```